pipeline/pipeline_test.go

```diff
@@ -0,0 +1,213 @@
+		ASN:              "AS10774x",
+		Location:         "nauscalosangeles",
+		Severity:         0.321565,
+		NumTestsAffected: 68089,
+		GoodPeriodInfo:   "Average download speed: 7.86 mb/s",
+		BadPeriodInfo:    "Average download speed: 5.33 mb/s",
+		IncidentInfo:     "Download speed dropped by 32.16% affecting 68089 tests",
+	}
+	ocTest := incident.IncidentJsonData{
+		GoodPeriodStart:  time.Date(2013, time.May, 1, 0, 0, 0, 0, time.UTC),
+		GoodPeriodEnd:    time.Date(2014, time.May, 1, 0, 0, 0, 0, time.UTC),
+		BadPeriodStart:   time.Date(2014, time.May, 1, 0, 0, 0, 0, time.UTC),
+		BadPeriodEnd:     time.Date(2017, time.August, 1, 0, 0, 0, 0, time.UTC),
+		GoodPeriodMetric: 6.335732,
+		BadPeriodMetric:  4.615034,
+		ASN:              "AS10143",
+		Location:         "ocau",
+		Severity:         0.546351,
+		NumTestsAffected: 30120,
+		GoodPeriodInfo:   "Average download speed: 6.34 mb/s",
+		BadPeriodInfo:    "Average download speed: 4.62 mb/s",
+		IncidentInfo:     "Download speed dropped by 54.64% affecting 30120 tests",
+	}
+	saTest := incident.IncidentJsonData{
+		GoodPeriodStart:  time.Date(2012, time.February, 1, 0, 0, 0, 0, time.UTC),
+		GoodPeriodEnd:    time.Date(2013, time.February, 1, 0, 0, 0, 0, time.UTC),
+		BadPeriodStart:   time.Date(2013, time.February, 1, 0, 0, 0, 0, time.UTC),
+		BadPeriodEnd:     time.Date(2015, time.June, 1, 0, 0, 0, 0, time.UTC),
+		GoodPeriodMetric: 0.864799,
+		BadPeriodMetric:  0.580199,
+		ASN:              "AS22927",
+		Location:         "sa",
+		Severity:         0.348264,
+		NumTestsAffected: 15180,
+		GoodPeriodInfo:   "Average download speed: 0.86 mb/s",
+		BadPeriodInfo:    "Average download speed: 0.58 mb/s",
+		IncidentInfo:     "Download speed dropped by 34.83% affecting 15180 tests",
+	}
+
+	// Creating a Google Cloud Storage client
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx, option.WithoutAuthentication())
+	if err != nil {
+		t.Errorf("Could not create GCS client")
+	}
+	// Obtain a bucket handle
+	bkt := client.Bucket(BUCKET_NAME)
+
+	tests := []struct {
+		name  string
+		input string
+		want  incident.IncidentJsonData
+	}{
+		{
+			name:  "Confirm incident in 'af' directory",
+			input: "af/ke/AS15399.json",
+			want:  afTest,
+		},
+		{
+			name:  "Confirm incident in 'as' directory",
+			input: "as/jp/AS2497.json",
+			want:  asTest,
+		},
+		{
+			name:  "Confirm incident in 'eu' directory",
+			input: "eu/AS1136.json",
+			want:  euTest,
+		},
+		{
+			name:  "Confirm incident in 'na' directory",
+			input: "na/us/ca/losangeles/AS10774x.json",
+			want:  naTest,
+		},
+		{
+			name:  "Confirm incident in 'oc' directory",
+			input: "oc/au/AS10143.json",
+			want:  ocTest,
+		},
+		{
+			name:  "Confirm incident in 'sa' directory",
+			input: "sa/AS22927.json",
+			want:  saTest,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Uncomment this call to update incidents on Google Cloud Storage
+			// runPipeline()
```

Comment:
Good tests. My understanding is that this would work just fine if left uncommented out all the time

Reply:
Works the same locally, but not on Travis. The reason that it is commented out is that (unless the bucket contents are deleted) the pipeline will not have to be run again in order for these tests to pass.